src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import {
+  Content,
+  FiatRow,
+  Aligner,
+  InputRow,
+  LabelRow,
+  Container,
+  InputPanel,
+  CurrencySelect,
+  StyledDropDown,
+  StyledTokenName,
+  UppercaseHelper,
+} from './CurrencyInputPanel.styles'
+import Loader from '../Loader'
+import { TYPE } from '../../theme'
+import { RowBetween } from '../Row'
+import DoubleCurrencyLogo from '../DoubleLogo'
+import NumericalInput from '../Input/NumericalInput'
+import { FiatValueDetails } from '../FiatValueDetails'
+import { CurrencyLogo, CurrencyWrapperSource } from '../CurrencyLogo'
+import { CurrencySearchModalComponent } from '../SearchModal/CurrencySearchModal'
+
+import { useActiveWeb3React } from '../../hooks'
+import { useCurrencyBalance } from '../../state/wallet/hooks'
+import { limitNumberOfDecimalPlaces } from '../../utils/prices'
+
+import { CurrencyInputPanelProps } from './CurrencyInputPanel.types'
+
+export const CurrencyInputPanelComponent = ({
+  id,
+  pair = null, // used for double token logo
+  onMax,
+  value,
+  label,
+  balance,
+  currency,
+  disabled,
+  hideInput = false,
+  isLoading = false,
+  fiatValue,
+  hideBalance = false,
+  priceImpact,
+  onUserInput,
+  otherCurrency,
+  chainIdOverride,
+  showCommonBases,
+  onCurrencySelect,
+  customBalanceText,
+  isFallbackFiatValue,
+  currencyWrapperSource = CurrencyWrapperSource.SWAP,
+  disableCurrencySelect = false,
+}: CurrencyInputPanelProps) => {
+  const { t } = useTranslation()
+
+  const [isOpen, setIsOpen] = useState(false)
+  const [focused, setFocused] = useState(false)
+  const { account } = useActiveWeb3React()
+
+  const selectedCurrencyBalance = useCurrencyBalance(account ?? undefined, currency ?? undefined)
+
+  const onDismiss = useCallback(() => {
+    setIsOpen(false)
+  }, [setIsOpen])
```

Comment:
```suggestion
  const onDismiss = useCallback(() => {
    setIsOpen(false)
  }, [])
```

React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.

Reply:
Fixed. You can take a look again :)